components/operator/internal/modules/module.go

```diff
@@ -114,26 +147,19 @@ func (r *moduleReconciler) installModule(ctx context.Context, registeredModules
 		}
 	}
 
-	var (
-		chosenVersion      Version
-		chosenVersionLabel string
-	)
-	for _, version := range sortedVersions(r.module) {
-		if !r.Versions.IsHigherOrEqual(r.module.Name(), version) {
-			break
-		}
-		chosenVersion = r.module.Versions()[version]
-		chosenVersionLabel = version
-		if chosenVersion.PreUpgrade == nil {
-			continue
+	for _, version := range sortedVersions {
+		if chosenVersion.PreUpgrade != nil {
```

Comment:
version instead of chosenVersion ?

Reply:
No more check if IsHigherOrEqual here also ?

Reply:
The check is made earlier. The 
```
if version == chosenVersionLabel {
			break
		}
```
prevent to run undesirable migration.
BUt you're right for chosenVersion / version. I change.